components/camera/camera.go

```diff
@@ -75,8 +77,78 @@ type Properties struct {
 
 // NamedImage is a struct that associates the source from where the image came from to the Image.
 type NamedImage struct {
-	Image      image.Image
+	data       []byte
+	img        image.Image
 	SourceName string
+	mimeType   string
+}
+
+// NamedImageFromBytes constructs a NamedImage from a byte slice, source name, and mime type.
+func NamedImageFromBytes(data []byte, sourceName, mimeType string) (NamedImage, error) {
+	if data == nil {
+		return NamedImage{}, fmt.Errorf("must provide image bytes to construct a named image from bytes")
+	}
+	if mimeType == "" {
+		return NamedImage{}, fmt.Errorf("must provide a mime type to construct a named image")
+	}
+	return NamedImage{data: data, SourceName: sourceName, mimeType: mimeType}, nil
+}
+
+// NamedImageFromImage constructs a NamedImage from an image.Image, source name, and mime type.
+func NamedImageFromImage(img image.Image, sourceName, mimeType string) (NamedImage, error) {
+	if img == nil {
+		return NamedImage{}, fmt.Errorf("must provide image to construct a named image from image")
+	}
+	if mimeType == "" {
+		return NamedImage{}, fmt.Errorf("must provide a mime type to construct a named image")
+	}
+	return NamedImage{img: img, SourceName: sourceName, mimeType: mimeType}, nil
+}
+
+// Image returns the image.Image of the NamedImage.
+func (ni *NamedImage) Image(ctx context.Context) (image.Image, error) {
+	if ni.img == nil {
```

Comment:
I think it could be clearer if you immediately returned `ni.img` if it is not nil at the beginning here

Reply:
Action taken here https://github.com/viamrobotics/rdk/pull/5183/commits/a90370f8989a55438f8a3488f95079bf42cd8677